sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BooleanSimplificationSuite.scala

```diff
@@ -126,6 +126,54 @@ class BooleanSimplificationSuite extends PlanTest with ExpressionEvalHelper with
       'a === 'b || 'b > 3 && 'a > 3 && 'a < 5)
   }
 
+  test("SPARK-34222: (a && b) && a && (a && c) => a && b && c") {
```

Comment:
nit: `SPARK-34222: simplify conjunctive predicates`

Reply:
I had kept it so to be consistent with other test names in the file.
changed to `test("SPARK-34222: simplify conjunctive predicates (a && b) && a && (a && c) => a && b && c")`